### PR TITLE
Task1/core concepts

### PR DESCRIPTION
**Pull request by prateekd15**
## Purpose
Task 1: Core concepts

## Screenshot
Counter, SearchForm, GenreSelect is rendered on the homescreen.
Counter and SearchForm accept and render initial values.
 
image

### CODE REVIEW

File: src/component/Counter/Counter.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			),
+			React.createElement(
+				"div",
+				{ className: "counter__btns" },
+				React.createElement(
+					"button",
+					{
+						onClick: this.handleDecrement,
+						className: "counter__btn",
+					},
```
Comment (Ajay-Nallanagula) [inline src/component/Counter/Counter.js]: Move button labels to *constant.js files

Comment (Ajay-Nallanagula) [reply src/component/Counter/Counter.js]: Suggestion Only :  
You have two buttons "+" and "-" with almost same code structure, try making it generic 
`createButton({label,callback,className}){  /* blah blah*/  }

//usage
[{label:'+', callback:this.handleDecrement, className:'counter__btn', }].map(createButton)
`

File: src/component/GenreSelect/GenreSelect.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import React from 'react';
+import './GenreSelect.css'; // Import your CSS file
+
+function GenreSelect({ genres, selectedGenre, onSelect }) {
+	const handleChange = (event) => {
+		const selectedGenre = event.target.value;
+		onSelect(selectedGenre);
+	};
+
+	const genreButtons = genres.map((genre) => {
```
Comment (Ajay-Nallanagula) [inline src/component/GenreSelect/GenreSelect.js:7]: I recommend , to write functional component(dumb/presentational component), for GenreButtons ==> GenreButton and then GenreSelect can be used as container component

**Review by Ajay-Nallanagula — CHANGES_REQUESTED** [review]

File: src/component/Counter/Counter.js (hunk 1 above)
Comment (prateekd15) [reply src/component/Counter/Counter.js]: Fixed as part of 53de1b8 commit

File: src/component/GenreSelect/GenreSelect.js (hunk 2 above)
Comment (prateekd15) [reply src/component/GenreSelect/GenreSelect.js:7]: Fixed as part of 53de1b8 commit

**Review by Ajay-Nallanagula — APPROVED** [review]

**Pull request merged**